zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/group/DbGroupState.java

```diff
@@ -126,6 +126,28 @@ public void delete(final long groupKey) {
     groupColumnFamily.deleteExisting(this.groupKey);
   }
 
+  @Override
+  public void addTenant(final long groupKey, final long tenantKey) {
+    this.groupKey.wrapLong(groupKey);
+    final PersistedGroup persistedGroup = groupColumnFamily.get(this.groupKey);
+    if (persistedGroup != null) {
```

Comment:
🔧 We should remove the condition. The processor has already checked that the group exists. If someone ends up trying to add a tenant to a group somewhere and the group doesn't exist, it's compeltely fair to encounter an exception. That means the caller hasn't been thorough with the request and didn't validate the request properly.